internal/driver/glfw/window.go

```diff
@@ -398,8 +402,11 @@ func (w *window) Content() fyne.CanvasObject {
 }
 
 func (w *window) SetContent(content fyne.CanvasObject) {
+	w.viewLock.RLock()
+	visible := w.visible
+	w.viewLock.RUnlock()
 	// hide old canvas element
-	if w.visible && w.canvas.Content() != nil {
+	if visible && w.canvas.Content() != nil {
```

Comment:
should w.canvas.Content also be captured in a local variable?

Reply:
Canvas does it's own locking

Reply:
Ok so Content() should be safe, perhaps capture the canvas reference then?

Reply:
Also line 354 accesses the canvas content without going through canvas' locking